project/Subject.cs

```diff
@@ -1,67 +1,60 @@
-namespace assignment_3;
-
-public class Subject
+namespace assignment_3
 {
-    public string SubjectId { get; private set; }
-    private string _subjectName;
-    public string SubjectName
+    public class Subject
     {
-        get => _subjectName;
-        private set
-        {
-            if (string.IsNullOrWhiteSpace(value))
-                throw new ArgumentException("Subject name cannot be null or empty.");
-            _subjectName = value;
-        }
-    }
+        public string SubjectId { get; private set; }
 
-    private int _gradingScale;
-    public int GradingScale
-    {
-        get => _gradingScale;
-        private set
+        private string _subjectName;
+        public string SubjectName
         {
-            if (value < 1 || value > 5)
-                throw new ArgumentException("Grading scale must be between 1 and 5.");
-            _gradingScale = value;
+            get => _subjectName;
+            private set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                    throw new ArgumentException("Subject name cannot be null or empty.");
+                _subjectName = value;
+            }
         }
-    }
-
-    public Subject(string subjectId, string subjectName, int gradingScale)
-    {
-        if (string.IsNullOrWhiteSpace(subjectId))
-            throw new ArgumentException("Subject ID cannot be null or empty.");
-
-        SubjectId = subjectId;
-        SubjectName = subjectName;
-        GradingScale = gradingScale;
-        AddSubject(this);
-        SaveManager.SaveToJson(_subjectList, nameof(_subjectList));
-    }
-
-    private static List<Subject> _subjectList = new();
 
-    private static void AddSubject(Subject subject)
-    {
-        if (subject is null)
+        private int _gradingScale;
+        public int GradingScale
         {
-            throw new ArgumentException($"{nameof(subject)} cannot be null.");
+            get => _gradingScale;
+            private set
+            {
+                if (value < 1 || value > 5)
+                    throw new ArgumentException("Grading scale must be between 1 and 5.");
+                _gradingScale = value;
+            }
         }
 
-        if (_subjectList.Any(s => s.SubjectId == subject.SubjectId))
+        private Teacher _teacher;
```

Comment:
Teacher <=> Students is 1..* <=> 1..*, meaning you can't have just the field Teacher, but you need the List<Teacher>, just like how you have the List<Subjects> in Teacher